WireAPI/Sources/WireAPI/APIs/FeatureConfigsAPI/FeatureConfigsAPIV4.swift

```diff
@@ -0,0 +1,164 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class FeatureConfigsAPIV4: FeatureConfigsAPIV3 {
+
+    override var apiVersion: APIVersion {
+        .v4
+    }
+
+    override func getFeatureConfigs() async throws -> [FeatureConfig] {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/feature-configs",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok, type: FeatureConfigsResponseAPIV4.self)
+            .failure(code: .forbidden, label: "operation-denied", error: FeatureConfigsAPIError.insufficientPermissions)
+            .failure(code: .forbidden, label: "no-team-member", error: FeatureConfigsAPIError.userIsNotTeamMember)
+            .failure(code: .notFound, label: "no-team", error: FeatureConfigsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+}
+
+struct FeatureConfigsResponseAPIV4: Decodable, ToAPIModelConvertible {
+    let appLock: FeatureWithConfig<FeatureConfigResponse.AppLockV0>
+    let classifiedDomains: FeatureWithConfig<FeatureConfigResponse.ClassifiedDomainsV0>
+    let conferenceCalling: FeatureWithoutConfig
+    let conversationGuestLinks: FeatureWithoutConfig
+    let digitalSignatures: FeatureWithoutConfig
+    let fileSharing: FeatureWithoutConfig
+    let selfDeletingMessages: FeatureWithConfig<FeatureConfigResponse.SelfDeletingMessagesV0>
+    let mls: FeatureWithConfig<FeatureConfigResponse.MLSV4>?
+    let mlsMigration: FeatureWithConfig<FeatureConfigResponse.MLSMigrationV4>? /// Starting api v4
+    let mlsE2EId: FeatureWithConfig<FeatureConfigResponse.EndToEndIdentityV4>? /// Starting api v4
+
+    func toAPIModel() -> [FeatureConfig] {
+        var featureConfigs: [FeatureConfig] = []
+
+        let appLockConfig = AppLockFeatureConfig(
+            status: appLock.status,
+            isMandatory: appLock.config.enforceAppLock,
+            inactivityTimeoutInSeconds: appLock.config.inactivityTimeoutSecs
```

Comment:
One way you could avoid this duplication is to make `FeatureWithConfig<FeatureConfigResponse.AppLockV0>` conform to `ToAPIModelConvertible` where the model is `AppLockFeatureConfig`. This of course would also apply to all other features with configs.

Reply:
Good point, thanks 👍

Reply:
done for feature with configs that don't change across different API versions.